src/libprojplugin/libprojplugin.cpp

```diff
@@ -104,13 +146,75 @@ QString LibprojPlugin::readProjectFile()
 bool LibprojPlugin::parseMetadata(const QString & strJson)
 {
     if (strJson.isNull() || strJson.isEmpty()) {
-        qWarning() << "Nothing to parse.";
+        qWarning() << "Nothing to parse";
         return false;
     }
     else {
-        bool ok = false;
-        parsedMetadata = QtJson::parse(strJson, ok).toMap();
-        Q_ASSERT(ok);
-        return true;
+        std::string err;
+        qDebug() << "About to start parsing";
+        projectData = Json::parse(strJson.toStdString(), err);
+        if (!err.empty())
+            qWarning() << "Something wrong with parsing:\n" + QString(err.c_str());
+        return err.empty() ? true : false;
     }
 }
+
+void LibprojPlugin::triggerAddNewFileAction()
+{
+   switch(QMessageBox::question(nullptr, tr("File exists?"),tr("Are you want to add existing file (Yes) or create new (No)?"),
+                                QMessageBox::Yes, QMessageBox::No))
+   {
+   case QMessageBox::StandardButton::Yes:
+   {
+       /* TODO
+        * ...
+        */
+       break;
+   }
+   case QMessageBox::StandardButton::No:
+   {
+       QString newFilename (QFileDialog::getSaveFileName(nullptr, tr("Where you want to save new file?"), project->document()->filePath().toFileInfo().dir().path()));
+       QFile newFile (newFilename);
+       if (!newFile.exists())
+           newFile.open(QIODevice::ReadWrite | QIODevice::Text);
+       else {
+           qWarning() << "File already exists!";
```

Comment:
What about return error code, or just false for example? You don't split gui and logis so you have such problems